src/lib.rs

```diff
@@ -29,6 +29,18 @@ impl Heap {
         }
     }
 
+    /// Initializes an empty heap
+    ///
+    /// # Unsafety
+    ///
+    /// This function must be called at most once and must only be used on an
+    /// empty heap.
+    pub unsafe fn init(&mut self, heap_bottom: usize, heap_size: usize) {
+        self.bottom = heap_bottom;
+        self.size = heap_size;
+        self.holes = HoleList::new(heap_bottom, heap_size);
```

Comment:
wouldn't it be cleaner to simply say `*self = Self::new(heap_bottom, heap_size);` ?